Raven.NET.Core/Observers/RavenTypeWatcher.cs

```diff
@@ -1,16 +1,120 @@
 using System;
+using System.Collections.Concurrent;
+using System.Collections.Generic;
+using System.Linq;
+using Raven.NET.Core.Configuration;
 using Raven.NET.Core.Observers.Interfaces;
+using Raven.NET.Core.Providers.Interfaces;
+using Raven.NET.Core.Static;
 using Raven.NET.Core.Subjects;
 
 namespace Raven.NET.Core.Observers
 {
     /// <inheritdoc/>
     public class RavenTypeWatcher : IRavenTypeWatcher
     {
+        private readonly IRavenProvider _ravenProvider;
+        private readonly IRavenSettingsProvider _ravenSettingsProvider;
+
+        private RavenSettings _ravenSettings = new();
+        private List<RavenSubject> _watchedSubjects = new();
+        private Func<RavenSubject,bool> _updateAction;
+        private string _keyName;
+
+        public RavenTypeWatcher(IRavenProvider ravenProvider, IRavenSettingsProvider ravenSettingsProvider)
+        {
+            _ravenProvider = ravenProvider;
+            _ravenSettingsProvider = ravenSettingsProvider;
+
+            _ravenSettings.BackgroundWorker = true;
+            _ravenSettings.BackgroundWorkerInterval = 1.0f;
+        }
+        
         /// <inheritdoc/>
         void IRaven.Update(RavenSubject subject)
         {
-            throw new NotImplementedException();
+            _updateAction.Invoke(subject);
+        }
+
+        /// <inheritdoc/>
+        public IRavenTypeWatcher Create<T>(string name, string keyName, Func<RavenSubject, bool> callback, Action<RavenSettings> options = null) 
+        {
+            _updateAction = callback;
+            _ravenProvider.AddRaven(name,this, typeof(T));
+            RavenCache.SubjectTypeCache.TryAdd(typeof(T), new ConcurrentDictionary<string, string>());
+            _keyName = keyName;
+
+            var ravenSettings = _ravenSettingsProvider.GetRaven(name);
+            if (ravenSettings != null)
+            {
+                if (ravenSettings.BackgroundWorker == null)
+                {
+                    ravenSettings.BackgroundWorker = true;
+                    ravenSettings.BackgroundWorkerInterval = 1.0f;
+                }
+                
+                _ravenSettings = ravenSettings;
+            }
+
+            if (options != null)
```

Comment:
i would recommend to use new c# syntax, ravenSettings is not null or ravenSettings is {}

Reply:
With Piotr.Ś at beginning we decided to use old fasion way, so i will keep this for consistency